pennylane/_queuing_context.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the :class:`QueuingContext` abstract base class.
+"""
+import abc
+
+
+class QueuingContext(abc.ABC):
+    """Abstract base class for classes that exposes a queue for Operations.
+
+    In PennyLane, quantum gates are not attached to some kind of circuit object
+    by the user. Here, the *construction* of a quantum gate is used to realize this
+    behaviour. As the constructor does not know to which QNode (PennyLane's circuit
```

Comment:
A suggestion here (subjective, feel free to discard it): I would suggest rewording the first sentence as follows (not too sure about the second sentence and later the third might have to be rephrased since the `QNode` was introduced there).
```suggestion
    In PennyLane, the construction of quantum gates is separated from the specific
    quantum node (:class:`BaseQNode`) that they belong to. As the
    constructor does not know to which QNode (PennyLane's circuit
```

Reply:
Sorry, I already merged because you approved and didn't write anything further -- I like that wording better though.